client/asset/eth/rpcclient.go

```diff
@@ -259,53 +257,13 @@ func (c *rpcclient) initiate(txOpts *bind.TransactOpts, netID int64, refundTimes
 }
 
 // initiateGas checks the amount of gas that is used for a call to the initiate function.
-func (c *rpcclient) initiateGas(ctx context.Context, refundTimestamp int64, secretHash [32]byte, participant *common.Address, contractAddress *common.Address) (uint64, error) {
-	parsedAbi, err := abi.JSON(strings.NewReader(swap.ETHSwapABI))
-	if err != nil {
-		return 0, err
-	}
-	data, err := parsedAbi.Pack("initiate", big.NewInt(refundTimestamp), secretHash, participant)
-	if err != nil {
-		return 0, err
-	}
-	msg := ethereum.CallMsg{
-		From:  *participant,
-		To:    contractAddress,
-		Gas:   0,
-		Value: big.NewInt(1),
-		Data:  data,
-	}
-	gas, err := c.ec.EstimateGas(ctx, msg)
-	if err != nil {
-		return 0, err
-	}
-
-	return gas, nil
+func (c *rpcclient) initiateGas(ctx context.Context, msg ethereum.CallMsg) (uint64, error) {
+	return c.ec.EstimateGas(ctx, msg)
 }
 
 // redeemGas checks the amount of gas that is used for a call to the redeem function.
-func (c *rpcclient) redeemGas(ctx context.Context, secret, secretHash [32]byte, participant *common.Address, contractAddress *common.Address) (uint64, error) {
-	parsedAbi, err := abi.JSON(strings.NewReader(swap.ETHSwapABI))
-	if err != nil {
-		return 0, err
-	}
-
-	data, err := parsedAbi.Pack("redeem", secret, secretHash)
-	if err != nil {
-		return 0, err
-	}
-	msg := ethereum.CallMsg{
-		From: *participant,
-		To:   contractAddress,
-		Gas:  0,
-		Data: data,
-	}
-	gas, err := c.ec.EstimateGas(ctx, msg)
-	if err != nil {
-		return 0, err
-	}
-
-	return gas, nil
+func (c *rpcclient) redeemGas(ctx context.Context, msg ethereum.CallMsg) (uint64, error) {
+	return c.ec.EstimateGas(ctx, msg)
 }
```

Comment:
Now that both of these functions are the same, I think you should combine them to a `*rpcclient.estimateGas` function.

Reply:
Yeah, good point. @martonp We'll merge this today if you merge these guys

Reply:
Done. This is nicer, good call.